web-common/src/features/dashboards/pivot/pivot-table-transformations.ts

```diff
@@ -127,14 +128,56 @@ export function getTotalsRowSkeleton(
 
     totalsRow = totalsRowTable[0] || {};
 
-    if (anchorDimensionName) {
+    if (anchorDimensionName && !config.isFlat) {
       totalsRow[anchorDimensionName] = "Total";
+    } else if (config.isFlat && anchorDimensionName) {
+      // For flat tables, find the first dimension column to place the Totals label
+      const firstDimensionName = getFirstDimensionForFlat(config);
+      if (firstDimensionName) {
+        totalsRow[firstDimensionName] = "Total";
+      }
     }
   }
 
   return totalsRow;
 }
 
+/**
+ * For flat tables, find the first dimension column to place the Totals label.
+ * This handles the case where measures might come before dimensions in the column order.
+ */
+function getFirstDimensionForFlat(config: PivotDataStoreConfig): string | null {
+  const { rowDimensionNames, pivot } = config;
+
+  // Go through the columns in order and find the first dimension
+  for (const column of pivot.columns) {
+    if (
+      column.type === PivotChipType.Dimension ||
+      column.type === PivotChipType.Time
+    ) {
+      // For time dimensions, we need to construct the actual dimension name
+      if (column.type === PivotChipType.Time) {
+        const timeDimension = config.time?.timeDimension;
+        if (timeDimension) {
+          const timeDimensionName = `${timeDimension}_rill_${column.id}`;
+          // Check if this time dimension is in our row dimensions
+          if (rowDimensionNames.includes(timeDimensionName)) {
+            return timeDimensionName;
+          }
+        }
+      } else {
+        // Regular dimension
+        if (rowDimensionNames.includes(column.id)) {
+          return column.id;
+        }
+      }
```

Comment:
This doesn't seem like the most optimal way of getting the first dimension. If we can clarify what to do in cases where the first column is not a dimension, I can refactor this part.